lib/modules/manager/npm/post-update/yarn.ts

```diff
@@ -103,12 +109,21 @@ export async function generateLockFile(
     const isYarnModeAvailable =
       minYarnVersion && semver.gte(minYarnVersion, '3.0.0');
 
-    let installYarn = 'npm i -g yarn';
-    if (isYarn1 && minYarnVersion) {
-      installYarn += `@${quote(yarnCompatibility)}`;
-    }
+    const preCommands: string[] = [];
+
+    const yarnTool: ToolConstraint = {
+      toolName: 'yarn',
+      constraint: '^1.22.18', // needs to be a v1 yarn, otherwise v2 will be installed
+    };
 
-    const preCommands = [installYarn];
+    if (config.managerData?.hasPackageManager) {
```

Comment:
@rarkins should we fallback to yarn v1 install when `isYarn1===true`, so we can use the `slim` (patched) yarn?

Reply:
Yes, that would be very desirable. It saves a lot of time and bandwidth